### PR TITLE
Fix - Choose reference over pointer

### PR DESCRIPTION
**Pull request by hamishwillee**
This fixes up docs to match PR https://github.com/dronecore/DroneCore/pull/316 (ie addresses #106 )

@shakthi-prashanth-m Can you please sanity check this.

### CODE REVIEW

File: en/api_reference/classdronecore_1_1_device.md [hunk 1]
```diff
@@ -81,15 +81,15 @@ uint8_t | [get_own_component_id](#classdronecore_1_1_device_1aa6dc1543b0ae9439a6
...
 ```cpp
-dronecore::Device::Device(DroneCoreImpl *parent, uint8_t target_system_id)
+dronecore::Device::Device(DroneCoreImpl &parent, uint8_t target_system_id)
 ```
 
 
 **Parameters**
 
-* DroneCoreImpl * **parent** - 
+* DroneCoreImpl & **parent** -
```
Comment (shakthi-prashanth-m) [inline en/api_reference/classdronecore_1_1_device.md]: Is there a space between `&` and **parent** ?
I think `DroneCoreImpl &parent` looks better than `DroneCoreImpl & parent`. What do you think ?

File: en/api_reference/classdronecore_1_1_follow_me.md [hunk 2]
```diff
@@ -67,12 +67,12 @@ Constructor. Creates the plugin for a specific [Device](classdronecore_1_1_devic
...
 
 ```cpp
-auto follow_me = std::make_shared<FollowMe>(&device);
+auto follow_me = std::make_shared<FollowMe>(device);
 ```
 
 **Parameters**
 
-* [Device](classdronecore_1_1_device.md) * **device** - The specific device associated with this plugin.
+* [Device](classdronecore_1_1_device.md) & **device** - The specific device associated with this plugin.
```
Comment (shakthi-prashanth-m) [inline en/api_reference/classdronecore_1_1_follow_me.md]: How about [Device](classdronecore_1_1_device.md) &**device** ?

File: en/examples/follow_me.md [hunk 3]
```diff
@@ -164,9 +164,9 @@ int main(int, char **)
 
     // Device got discovered.
     Device &device = dc.device();
-    std::shared_ptr<Action> action = std::make_shared<Action>(&device);
-    std::shared_ptr<FollowMe> follow_me = std::make_shared<FollowMe>(&device);
-    std::shared_ptr<Telemetry> telemetry = std::make_shared<Telemetry>(&device);
+    std::shared_ptr<Action> action = std::make_shared<Action>(device);
```
Comment (shakthi-prashanth-m) [inline en/examples/follow_me.md]: Why not `auto action = std::make_shared<Action>(device);` ?

Comment (hamishwillee) [reply en/examples/follow_me.md]: No reason at all, except that this is a direct copy of your example code :-) 
https://github.com/dronecore/DroneCore/blob/develop/example/follow_me/follow_me.cpp#L53

So if the example changes, so will this.

Comment (shakthi-prashanth-m) [reply en/examples/follow_me.md]: oh! :) Sorry. I found my mistake!

File: en/api_reference/classdronecore_1_1_device.md (hunk 1 above)
Comment (hamishwillee) [reply en/api_reference/classdronecore_1_1_device.md]: This is what is generated by the toolchain (not this specific change), so IMO we should accept this PR and deal with the rendering as a separate issue.

I'll merge and lets carry on the conversation in https://github.com/dronecore/DroneCore/issues/321

**Pull request merged**